articles/anomaly-detection/passwords-checking.md

```diff
@@ -0,0 +1,229 @@
+4th line.
+
+### Starting the job
+Once you have the file ready, you are ready to start a job to check the file against your database connection. To do so you need to execute a `POST` request against `/api/passwords-checking` with encoding type `multipart/form-data`.
+
+Your request should contain the following parameters:
+
+- passwords (the csv file, detailed above, that you are uploading, which contains data to check)
+- connection_id (a string, the connection id of the connection from where the users will be checked)
+
+If it works, you will get a response similar to the following one:
+
+```js
+{
+    "id":"job_abcdef1234567890",
+    "status":"pending",
+    "type":"passwords_checking",
+    "connection":"abcd"
+}
+```
+
+The returned entity represents the passwords checking job. At this point the job is queued on our system and will be started
+as soon as possible. Once the job is finishes you will get a log entry `fpc` (if failed) or `spc` (if succeed). You can query the logs through [Management API2](/api/management/v2) or use our [Dashboard](${manage_url}/#/logs) to get them. You can also query job status anytime you want using the appropriate endpoint (see bellow).
+
+### Querying job status
+Once a job has been started you can query its status whenever you want, to do so simply make a `GET` request to `/api/jobs/{id}`, the `id` is the one that has been handed to you when you started the job (see above). You will get a response similar to
+
+```js
+{
+    "id":"job_abcdef1234567890",
+    "status":"pending",
+    "type":"passwords_checking",
+}
+```
+
+The `status` field might contain any of the following values:
+
+- pending: The job is still queued or being executed on our system. Wait some minutes and query again to know when it has finished.
+- completed: The job is complete you can query the errors and the results (see bellow).
+- failed: The job has failed, you must query the errors and check your logs to see what went wrong.
+
+### Querying results
+Once the job has finished you are ready to gets it results and act based on them. To get the results you should make a request to `/api/{job_id}/results`. You will get a JSON Array response, similar to:
+
+```js
+[
+  {
+    "email": "john.doe1@example.com",
+    "status": "password_matched"
+  },
+  {
+    "username": "john.doe2",
+    "status": "invalid_password"
+  },
+  {
+    "username": "john.doe2",
+    "status": "user_not_found"
+  }
+]
+```
+
+- Email or username will be available depending the information you provided as input.
+- Status might have any of the following values:
+  * password_matched: the password you provided for that user matched
+  * invalid_password: the password you provided haven't matched
+  * user_not_found: the user wasn't available in the connection database you specified.
+
+Keep in mind that we try to recover from most of the errors instead of failing the job, so even when the job is complete
```

Comment:
+Note that the system will recover from most of errors instead of failing the job, so there may be rows with errors even after the job completes. You should check the errors (see below) after every job.